[FEATURE] Include POIs into generated maps and diplay them with VTM rendering

treee111 opened the pull request:
## This PR…

- introduces POIs into the generated files and display them as symbol on your wahoo device!
  - bakery
  - drinking water
  - fuel station
  - cafe

## Considerations and implementations

- launch config for POIs
- adjustments to enable POIs
  - constants update to NOT-filter out POIs
  - tag-wahoo file update to include POIs into generated file
  - VTM device theme for POIs incl. symbols
- documentation adjustments

## How to test

1. create maps and act like written in the documentation

## Pull Request Checklist
- [x] Reviewed the [Contributing Guidelines](https://github.com/treee111/wahooMapsCreator/blob/develop/.github/CONTRIBUTING.md)
    + Especially the [Pull-Requests](https://github.com/treee111/wahooMapsCreator/blob/develop/.github/CONTRIBUTING.md#Pull-Requests) section
- [ ] Commits (and commit-messages) are understandable
- [x] Tested with macOS / Linux
- [x] Tested with Windows

zenziwerken committed:
Added some tags not to be filtered.

Added landuse forest and amenity for =fuel =cafe, shop=bicycle =bakery

treee111 committed:
add docu for POI creation

treee111 committed:
copy files in

- from: https://github.com/zenziwerken/Bolt2-Mapsforge-Rendertheme

treee111 committed:
initial VTM rendering theme

- from https://gist.github.com/Intyre/7494d5c6c86b5bc60ce4d03b9cfce2df

treee111 committed:
write new files to the created .zip files

treee111 committed:
move themes folder to root directory

- docs: copy the whole vtm_theme_poi content (does not harm on e.g. BOLT)

treee111 committed:
scale icons to appear bigger

treee111 committed:
launch config for POIs

treee111 committed:
read out content of generated .map files

- to check, if certain OSM tags / POIs are in the generated .map files

mweirauch commented:
I think you don't need any custom symbols in the project. They are part of the vtm library (to my understanding).

https://github.com/mapsforge/vtm/tree/master/vtm-themes/resources/assets

Using a @Intyre inspired snippet in my custom vtm-elemnt.xml shows up POI icons just fine in Cruiser and on the Bolt2:
(of course with a modified tag-mapping to include these tags and custom keep-tag-constants, but the latter was merged ...)

```xml
	<m k="amenity" zoom-min="15">
		<m v="cafe">
			<symbol src="assets:symbols/food/cafe.svg" symbol-percent="100" />
		</m>
		<m v="drinking_water">
			<symbol src="assets:symbols/food/drinkingtap.svg" symbol-percent="100" />
		</m>
		<m v="fuel">
			<symbol src="assets:symbols/transport/fuel.svg" symbol-percent="100" />
		</m>
	</m>

	<m k="shop" zoom-min="15">
		<m v="bakery">
			<symbol src="assets:symbols/shopping/bakery.svg" />
		</m>
		<m v="bicycle">
			<symbol src="assets:symbols/shopping/bicycle.svg" />
		</m>
		<m v="supermarket">
			<symbol src="assets:symbols/shopping/supermarket.svg" />
		</m>
	</m>
```

mweirauch commented on .vscode/launch.json:
```diff
@@ -84,6 +84,42 @@
                 "-gt"
             ]
         },
+        {
```
IMHO no IDE specific files should be part of the repo. One usually does not spot any such files in other repositories. It's constantly clashing with my file when I am adding some quick launch snippets. I can deal with this, though ;)

treee111 replied:
what line do you mean in particular?
I'm with you that there should not (or only few) be OS specific files/settings in the repo.
For me, the launch config work in both macOS and Windows without a problem using VSCode.

treee111 replied:
read it a second time and you mean the whole file because it is IDE - VSCode - specific and would see the launch.json in the gitignore?
For me it is convenient if I have the same launch configs on both of my computers and additionally there is a launch config which is used as part of unittests.
Is is clashing because you change the launch.json and Git wants you to commit the changes?

mweirauch replied:
Yap, I meant the whole file. I know it is convenient, but only for one person working on the project. But no biggy. I'll just refrain from adding some on my own.

treee111 replied:
I got your point and this is valid in my opinion when many users develop a project. Since we are only a few active people and know how to get around it I would leave the file in.
In my opinion feel free to add entries here so we all have them.

treee111 committed:
add drinking_water POI in all relevant files

treee111 committed:
make some streets black for bolt

treee111 committed:
do not render place/locality

treee111 committed:
Merge branch 'release-v2.0.0' into enable-pois

treee111 committed:
Merge branch 'release-v2.0.0' into enable-pois

treee111 committed:
publish device_themes as separate .zip
- refactor and correct copyFiles  script
- correct cd .. statements

treee111 committed:
fix unittests according to new POI constants

treee111 committed:
adjust documentation

treee111 commented:
> I think you don't need any custom symbols in the project. They are part of the vtm library (to my understanding).
> 
> https://github.com/mapsforge/vtm/tree/master/vtm-themes/resources/assets
> 
> Using a @Intyre inspired snippet in my custom vtm-elemnt.xml shows up POI icons just fine in Cruiser and on the Bolt2: (of course with a modified tag-mapping to include these tags and custom keep-tag-constants, but the latter was merged ...)
> 
> ```
> 	<m k="amenity" zoom-min="15">
> 		<m v="cafe">
> 			<symbol src="assets:symbols/food/cafe.svg" symbol-percent="100" />
> 		</m>
> 		<m v="drinking_water">
> 			<symbol src="assets:symbols/food/drinkingtap.svg" symbol-percent="100" />
> 		</m>
> 		<m v="fuel">
> 			<symbol src="assets:symbols/transport/fuel.svg" symbol-percent="100" />
> 		</m>
> 	</m>
> 
> 	<m k="shop" zoom-min="15">
> 		<m v="bakery">
> 			<symbol src="assets:symbols/shopping/bakery.svg" />
> 		</m>
> 		<m v="bicycle">
> 			<symbol src="assets:symbols/shopping/bicycle.svg" />
> 		</m>
> 		<m v="supermarket">
> 			<symbol src="assets:symbols/shopping/supermarket.svg" />
> 		</m>
> 	</m>
> ```

Good point! I checked that on my BOLTv1. That did not work. In cruiser it also worked without having the symbols.
I think this is because only the BOLTv2 has VTM rendering per default and the VTM library on the device while the others don't (or have a version without some symbols etc.)

treee111 committed:
use new symbol for drinking water

- from: https://github.com/mapsforge/vtm/blob/master/vtm-themes/resources/assets/symbols/food/drinkingtap.svg

treee111 committed:
adjust launch config to correct module name

- remaning from wahoo_mc to mahoomc was via #117

treee111 committed:
vtm device theme adjustments

- make highways smaller on bolt (because they are black)
- who drinking_water on the same zoom-level as bakeries

treee111 committed:
adjust gitignore for osmfilter files

treee111 committed:
adjust copyFiles script

treee111 committed:
README: add POI pictures
